Scripts/Runtime/Entities/TaskSystem/AbstractTaskDriver.cs

```diff
@@ -16,14 +15,20 @@ public abstract class AbstractTaskDriver<TTaskSystem> : AbstractTaskDriver
     {
         /// <summary>
         /// Reference to the associated <typeparamref name="TTaskSystem"/>
-        /// Hides the base reference to the abstract version.
         /// </summary>
-        public new TTaskSystem TaskSystem
+        /// <remarks>
+        /// Hides the base reference to the abstract version. This is so that from the outside, a developer
+        /// doesn't try and select a DataStream to configure a job on.
+        /// </remarks>
+        /// TODO: #67 - Should add a Safety check on Job scheduling that we're allowed to write
+        /// TODO: #100 - OR we don't need this to be hidden because we're allowed to write to TaskSystem data.
+        protected new TTaskSystem TaskSystem
         {
             get => (TTaskSystem)base.TaskSystem;
         }
 
-        protected AbstractTaskDriver(World world) : base(world, typeof(TTaskSystem))
+        // MIKE: I can't think of a better pattern here.
```

Comment:
I need to pass a parent in here because I need to know if I'm a top level TaskDriver or if I was created as a SubTaskDriver. 

But this kind of sucks because then you have to do:

`new WanderTaskDriver(World, null);`

But I can't call a different constructor when building through the sub task driver path because that's a user land constructor.

Reply:
I have managed to fix this now in my current work. Publicly you can only create the constructor that doesn't take in a parent. 

With the `TaskDriverFactory`, when generating the sub TaskDrivers, it will call a generated version that sets the parent.